packages/invoices/src/InboundShipment/api/hooks/line/useDeleteSelectedLines.ts

```diff
@@ -1,48 +1,50 @@
 import {
   useTableStore,
   useTranslation,
-  useNotification,
+  useDeleteConfirmation,
 } from '@openmsupply-client/common';
 import { useIsInboundDisabled } from '../utils/useIsInboundDisabled';
 import { useDeleteInboundLines } from './useDeleteInboundLines';
 import { useInboundRows } from './useInboundRows';
 
-export const useDeleteSelectedLines = (): {
-  onDelete: () => Promise<void>;
-} => {
-  const { success, info } = useNotification();
+export const useDeleteSelectedLines = (): (() => void) => {
   const { items, lines } = useInboundRows();
-  const { mutate } = useDeleteInboundLines();
+  const { mutateAsync } = useDeleteInboundLines();
   const isDisabled = useIsInboundDisabled();
   const t = useTranslation('replenishment');
 
-  const selectedRows = useTableStore(state => {
-    const { isGrouped } = state;
+  const selectedRows =
+    useTableStore(state => {
+      const { isGrouped } = state;
 
-    return isGrouped
-      ? items
-          ?.filter(({ id }) => state.rowState[id]?.isSelected)
-          .map(({ lines }) => lines.flat())
-          .flat()
-      : lines?.filter(({ id }) => state.rowState[id]?.isSelected);
-  });
+      return isGrouped
+        ? items
+            ?.filter(({ id }) => state.rowState[id]?.isSelected)
+            .map(({ lines }) => lines.flat())
+            .flat()
+        : lines?.filter(({ id }) => state.rowState[id]?.isSelected);
+    }) || [];
```

Comment:
Thought it was better to have empty array fallback here rather than allowing it to be possibly `undefined` in delete confirmation.